### DISCUSSION TITLE
LERC Elevation Tile Support

### DISCUSSION THREAD

**Discussion by tech-maven-geospatial** [TerriaJS/terriajs#4890] [Q&A] [answered]
We are interested in integrating in ESRI LERC Elevation Tiles I know Cesium supports this
Has anyone tried supporting this with Terriajs?
https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer
https://elevation.arcgis.com/arcgis/rest/services/WorldElevation/Terrain/ImageServer
https://developers.arcgis.com/documentation/tiled-elevation-service/

https://cesium.com/blog/2019/06/25/arcgis-elevation-tiles/

We've created LERC tiles with ArcGIS Pro (TPKX and then used extract package and converted folder of LERC2 tiles to GPKG and MBTILES) So we can serve this from our tile server or use it offline in Earth Explorer 3D Map app http://earthexplorer.world

**Comment by rowanwins** [comment, marked as answer]
Hi @tech-maven-geospatial 

I've just created a PR for this
https://github.com/TerriaJS/terriajs/pull/4899

Although it seems like perhaps there is something upstream in Cesium that might need to be looked at when working with an ImageServer that is only shipping elevation values for a small area.

**Reply by techmavengeospatial** [reply]
It seems like it's possible to serve locally from a tile server or folder of tiles webserver
if this  URL http://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer/tile/{z}/{y}/{x}.lerc2
is available from ESRI as XYZ Tile URL
then I can't understand why a URL like http://someip:port/folder/tile/{z}/{y}/{x}.lerc2 could not be given to Terriajs/cesium.

https://github.com/CesiumGS/cesium/blob/3601fc73247ab3526c4f18be38ccc8c1860a541b/Source/Core/ArcGISTiledElevationTerrainProvider.js#

https://cesium.com/learn/cesiumjs/ref-doc/ArcGISTiledElevationTerrainProvider.html
https://github.com/CesiumGS/cesium/blob/main/Apps/Sandcastle/gallery/ArcGIS%20Tiled%20Elevation%20Terrain.html
The URL is mentioned in this resource doc
https://cesium.com/learn/cesiumjs/ref-doc/Resource.html

In my testing I've not been able to get it working in Cesium or Terriajs
I've created my own LERC tiles 
tried this tool
[GitHub - DaYeSquad/tif2lerc-cpp: Convert tiff to esri lerc format](https://github.com/DaYeSquad/tif2lerc-cpp) 
using ESRI's Python Code
[GitHub - Esri/lerc: Limited Error Raster Compression](https://github.com/Esri/lerc)
I can transform a folder of TIF tiles into LERC
I like this approach because it gives me more compatibility with ESRI ArcGIS runtime or ArcGIS API For JavaScript 
but it's not good if it must come from ImageServer
